lib/Controller/SAMLController.php

```diff
@@ -175,9 +181,22 @@ public function login($idp) {
 			case 'saml':
 				$auth = new Auth($this->SAMLSettings->getOneLoginSettingsArray($idp));
 				$ssoUrl = $auth->login(null, [], false, false, true);
-				$this->session->set('user_saml.AuthNRequestID', $auth->getLastRequestID());
-				$this->session->set('user_saml.OriginalUrl', $this->request->getParam('originalUrl', ''));
-				$this->session->set('user_saml.Idp', $idp);
+				$response = new Http\RedirectResponse($ssoUrl);
+
+				// Pack data as JSON so we can properly extract it later
+				$data = json_encode([
+					'AuthNRequestID' => $auth->getLastRequestID(),
+					'OriginalUrl' => $this->request->getParam('originalUrl', ''),
+					'Idp' => $idp
+				]);
+
+				// Encrypt it
+				$data = $this->crypto->encrypt($data);
+
+				// And base64 encode it
+				$data = base64_encode($data);
+
+				$response->addCookie('saml_data', $data, null, 'None');
```

Comment:
Note [to myself], before merging, we ought to branch off (and better switch to regular stable branches, as we have them in most other apps).

Reply:
I don't really see why.
The code here should work also work on older releases. else we end up with the backport mess for every release.

Reply:
Okay. I expected a Warning when the parameter count mismatches.

Reply:
ah mmm right...

Reply:
It does not complain when there are too many, only when there are too few. So it works. If awkward.